### PR TITLE
E2e test infra

### PR DESCRIPTION
**Pull request by alexec**
This attempts to improve the e2e infra by:

* Logging more output from commands.
* Adding a level of abstraction to simplify and enhance tests around app management.

### CODE REVIEW

File: test/e2e/fixtures/app/consequences.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	return c
+}
+
+func (c *Consequences) And(block func(app *Application)) *Consequences {
+	block(c.app())
+	return c
+}
+
+func (c *Consequences) When() *Actions {
+	return c.actionable
```
Comment (alexec) [inline test/e2e/fixtures/app/consequences.go]: rename

File: errors/errors.go [hunk 2]
```diff
@@ -10,3 +10,7 @@ func CheckError(err error) {
 		log.Fatal(err)
 	}
 }
+
+func CheckError2(_ interface{}, err error) {
```
Comment (alexmt) [inline errors/errors.go]: Can we avoid this `CheckError2` name please? I cannot propose a better name, so instead can you move it to `fixture.go` and rename to `checkError` ?

**Review by alexmt — APPROVED** [review]
Conditional LGTM. Please address one comment about `CheckError2` method name

File: errors/errors.go (hunk 2 above)
Comment (alexec) [reply errors/errors.go]: It's a rubbish name. I've been trying to think of something better, but nothing has come to mind.  So I'll do this.

Comment (alexec) [reply errors/errors.go]: How about `FailOnErr`?

Comment (alexmt) [reply errors/errors.go]: Sounds good!

**Comment by alexec** [comment]
```
--- FAIL: TestManipulateApplicationResources (32.80s)
--- FAIL: TestAppWithSecrets (4.70s)
```

**Comment by alexec** [comment]
```
--- FAIL: TestManipulateApplicationResources (32.78s)
--- FAIL: TestAppWithSecrets (4.78s)
```

**Pull request merged**